src/Microsoft.AspNetCore.OData/Common/TypeHelper.cs

```
@@ -40,7 +41,136 @@ public static bool IsDynamicTypeWrapper(this Type type)
 
         public static bool IsSelectExpandWrapper(this Type type, out Type entityType) => IsTypeWrapper(typeof(SelectExpandWrapper<>), type, out entityType);
 
-        public static bool IsComputeWrapper(this Type type, out Type entityType) => IsTypeWrapper(typeof(ComputeWrapper<>), type, out entityType);
+        /// <summary>
+        /// Determines whether the specified type is a <see cref="ComputeWrapper{T}"/> or a custom implementation
+        /// that inherits from <see cref="DynamicTypeWrapper"/> and implements both <see cref="IComputeWrapper{T}"/>
+        /// and <see cref="IGroupByWrapper{TContainer, TWrapper}"/>.
+        /// </summary>
+        /// <param name="typeToCheck">The type to check.</param>
+        /// <param name="entityType">The entity type if the specified type is a <see cref="ComputeWrapper{T}"/> or a custom implementation; otherwise, null.</param>
+        /// <returns>
+        /// <c>true</c> if the specified type is a <see cref="ComputeWrapper{T}"/> or a custom implementation
+        /// that meets the criteria; otherwise, <c>false</c>.
+        /// </returns>
+        public static bool IsComputeWrapper(this Type typeToCheck, out Type entityType)
+        {
+            entityType = null;
+            if (typeToCheck == null)
+            {
+                return false;
+            }
+
+            if (typeToCheck.IsGenericType)
+            {
+                Type genericTypeDefinition = typeToCheck.GetGenericTypeDefinition();
+
+                // Default implementation
+                if (genericTypeDefinition == typeof(ComputeWrapper<>))
+                {
+                    entityType = typeToCheck.GetGenericArguments()[0];
+
+                    Debug.Assert(
```

Is this only meant for debugging or we should throw an exception instead.

---

@ElizabethOkerio Because `ComputeWrapper<T>` is the library default/built-in implementation of the `IComputeWrapper<T>` interface, it's safe to `Debug.Assert` instead of throwing an exception. The logic for throwing an exception would be unreachable since there's no time we're going to have the `ComputeWrapper<T>` type not inheriting from the `DynamicTypeWrapper` type or not implementing the required interfaces.